planet/System.ocg/Legacy.c

```diff
@@ -21,6 +21,22 @@ global func GetMissionAccess(string password)
 	return GetScenarioAccess(password);
 }
 
+/* -- Player stuff -- */
+
+global func GetPlayerColor(int player_nr)
+{
+	LogLegacyWarning("GetPlayerColor", "Since 9.0 OC you can call GetPlayer(player)->GetColor() instead.", VERSION_10_0_OC);
+	var player = GetPlayer(player_nr);
+	if (player)
+	{
+		return player->GetColor();
+	}
+	else
+	{
+		return 0;
```

Comment:
With the new interface I don't like that there are no defaults for missing stuff anymore. The good side is that you can identify that there is no such player easily, because it will be nil. The bad side is that those scripts that relied on the default will now have to implement a nil check and return the old value... which leads to a lot of duplicated code in the worst case.

Reply:
I'm not sure how common it is to call these functions on invalid players. Of course, the wrapper for the legacy functions needs to handle this case, but I don't think it's an issue for everything else.